test/functional/hostgroups_controller_test.rb

```diff
@@ -201,5 +189,20 @@ def setup_user(operation, type = 'hostgroups')
       assert_equal "overridden", child.parameters["y"]
       assert_equal nil, child.parameters["x"]
     end
+
+    it "changes the hostgroup's parent and check the parameters are updated" do
```

Comment:
This only tests the update action, not the AJAX of the parent update, which is what you're changing here.  I'm happy for the new test, and also reasonably happy to leave this without a test - but ideally, it'd be a new integration test that changes the host group parent and verifies the parameters tab is updated.  Could you add that?